### ISSUE TITLE
Support deploy web app to slot

### ISSUE THREAD

**Issue by ifle** [Microsoft/azure-maven-plugins#151]
I try to deploy the web app to slot without any success. Is maven plugin can used to deploy to app service slot?
```
[ERROR] Failed to execute goal com.microsoft.azure:azure-webapp-maven-plugin:1.1.0:deploy (default-cli) on project myapp: The host name myapp-api-dev1.azurewebsite
s.net is already assigned to another Azure website: myapp-api(dev1).: OnError while emitting onNext value: retrofit2.Response.class -> [Help 1]
[ERROR] 
```

**Comment by jdneo** [comment]
Hi @ifle. Thank you for your feedback. Currently the maven plugin does not support deploy to app service slot. We will support it in the future.

**Comment by SummerSun** [comment]
We are working on this feature, and the deployment slot will be supported by configurations:
```
<deploymentSlotSetting>
    <slotName>slot</slotName>
    <configurationSource>test</configurationSource>
</deploymentSlotSetting>
```

One should make sure the configured web app exists before deploying to the deployment slot.
We will create a new deployment slot if it does not exists first before the deployment, or, directly deploy to it if it already exists.

The ```<slotName>``` is THE slot name, not the ```<webAppName>-<slotName>```, will add more these kind of details in README.


### List the work items:

- [ ] Implementation
  PR #232 PR #234 
- [ ] Add Unit Tests
- [ ] Add integration test for deployment slot
- [ ] Update ReadMe to add configuration guidance

**Comment by jdneo** [comment]
Closing this issue since the features are all merged into our dev branch.

@ifle We will inform you in this issue thread when the new version of plugin released into Maven Central.

**Comment by ifle** [comment]
Thanks

**Comment by SummerSun** [comment]
@ifle We have released webapp 1.3.0 with deployment slot support included. Please give it a try. Thank you.
